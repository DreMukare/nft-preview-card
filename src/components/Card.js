import '../styles/card.scss';
import equilibrium from '../images/image-equilibrium.jpg';
import iconEthereum from '../images/icon-ethereum.svg';
import iconView from '../images/icon-view.svg';
import iconClock from '../images/icon-clock.svg';
import avatar from '../images/image-avatar.png';

const Card = () => {
	return (
		<div className='container'>
			<div className='card'>
				<div className='card__image'>
					<img src={equilibrium} alt='nft preview' />
					<div className='overlay'>
						<img src={iconView} alt='eye' />
					</div>
				</div>
				<section className='card__content'>
					<h2>Equilibrium #3429</h2>
					<p>Our Equilibrium collection promotes balance and calm.</p>
				</section>
				<section className='card__detail'>
					<div className='price'>
						<section className='image'>
							<img src={iconEthereum} alt='Ethereum icon' />
						</section>
						<p>0.041ETH</p>
					</div>
					<div className='time'>
						<img src={iconClock} alt='clock icon' />
						<p>3 days left</p>
					</div>
				</section>
				<hr />
				<section className='card__attribution'>
					<img src={avatar} alt='Creator of NFT' />
					<p>
						Creation of <span>Jules Wyvern</span>
					</p>
				</section>
			</div>
		</div>
	);
};

export default Card;
