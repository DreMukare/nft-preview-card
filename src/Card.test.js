import { render, screen } from '@testing-library/react';
import Card from './components/Card';

it('matches snapshot', () => {
	const { asFragment } = render(<Card />);

	expect(asFragment()).toMatchSnapshot();
});

test('whether paragraph is rendered', () => {
	const { getByText } = render(<Card />);

	expect(
		getByText(/Our Equilibrium collection promotes balance and calm./i)
	).toBeInTheDocument();
});

test('whether creator mentioned in attribution', () => {
	const { getByText } = render(<Card />);

	expect(getByText(/Jules Wyvern/i)).toBeInTheDocument();
});
