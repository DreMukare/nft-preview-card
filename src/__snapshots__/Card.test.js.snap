// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card"
  >
    <div
      class="card__image"
    >
      <img
        alt="nft preview"
        src="image-equilibrium.jpg"
      />
      <div
        class="overlay"
      >
        <img
          alt="eye"
          src="icon-view.svg"
        />
      </div>
    </div>
    <section
      class="card__content"
    >
      <h2>
        Equilibrium #3429
      </h2>
      <p>
        Our Equilibrium collection promotes balance and calm.
      </p>
    </section>
    <section
      class="card__detail"
    >
      <div
        class="price"
      >
        <section
          class="image"
        >
          <img
            alt="Ethereum icon"
            src="icon-ethereum.svg"
          />
        </section>
        <p>
          0.041ETH
        </p>
      </div>
      <div
        class="time"
      >
        <img
          alt="clock icon"
          src="icon-clock.svg"
        />
        <p>
          3 days left
        </p>
      </div>
    </section>
    <hr />
    <section
      class="card__attribution"
    >
      <img
        alt="Creator of NFT"
        src="image-avatar.png"
      />
      <p>
        Creation of 
        <span>
          Jules Wyvern
        </span>
      </p>
    </section>
  </div>
</DocumentFragment>
`;
